peer/peer.go

```diff
@@ -0,0 +1,179 @@
+package peer
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	host "github.com/libp2p/go-libp2p-host"
+	net "github.com/libp2p/go-libp2p-net"
+	lpeer "github.com/libp2p/go-libp2p-peer"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+	swarm "github.com/libp2p/go-libp2p-swarm"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	ma "github.com/multiformats/go-multiaddr"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// DefaultPort is the TCP port hosts will communicate over if none is
+	// provided
+	DefaultPort = 8765
+
+	// CumulusProtocol is the name of the protocol peers communicate over
+	CumulusProtocol = "/cumulus/0.0.1"
+
+	// DefaultIP is the IP address new hosts will use if none if provided
+	DefaultIP = "127.0.0.1"
+)
+
+// Peer is a cumulus Peer composed of a host
+type Peer struct {
+	host.Host
+}
+
+// NewPeer creates a Cumulus host with the given IP addr and TCP port.
+// This may throw an error if we fail to create a key pair, a pid, or a new
+// multiaddress.
+func NewPeer(ip string, port int) (*Peer, error) {
```

Comment:
If you're making a package with one primary export (ie `Peer`) then just call the constructor `New`. Then you're calling `peer.New(...)` -- makes more sense semantically